contracts/governance/Governor.sol

```diff
@@ -248,8 +246,10 @@ abstract contract Governor is Context, ERC165, EIP712, IGovernor, IERC721Receive
         bytes[] memory calldatas,
         string memory description
     ) public virtual override returns (uint256) {
+        uint256 timepoint = clock();
+
         require(
-            getVotes(_msgSender(), block.number - 1) >= proposalThreshold(),
+            getVotes(_msgSender(), timepoint - 1) >= proposalThreshold(),
```

Comment:
There seems to be a missing property in EIP-5805 saying that `getPastVotes` should be defined over all integers less than the current clock.

Reply:
Right. We have
> For any timepoint that is strictly smaller than clock(), the value returned by getPastVotes should be constant.

but we should add a constraint that it should not revert!